File: src/main/java/com/hy/frame/view/LoadingDialog.java
package com.hy.frame.view;

import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.WindowManager;
import android.widget.TextView;

import com.hy.frame.R;
import com.hy.frame.common.BaseDialog;

/**
 * 加载对话框
 * author HeYan
 * time 2015/12/11 18:11
 */
public class LoadingDialog extends BaseDialog {

    private TextView txtLoadMsg;
    private String loadMsg;

    public LoadingDialog(Context context) {
        this(context, null);
    }

    public LoadingDialog(Context context, String loadMsg) {
        super(context);
        this.loadMsg = loadMsg;
    }

    @Override
    protected int initLayoutId() {
        return R.layout.dlg_loading;
    }

    @Override
    protected void initWindow() {
        windowDeploy(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT, Gravity.CENTER);
    }

    @Override
    protected void initView() {
        txtLoadMsg = getView(R.id.loading_txtLoadMsg);
    }

    @Override
    protected void initData() {
        if (loadMsg != null)
            txtLoadMsg.setText(loadMsg);
    }

    @Override
    protected void onViewClick(View v) {

    }

    public void updateMsg(String msg) {
        if (txtLoadMsg != null) {
            txtLoadMsg.setText(msg);
        } else {
            loadMsg = msg;
        }
    }

}

File: src/main/java/com/hy/frame/ui/LoadingDialog.kt
package com.hy.frame.ui

import android.content.Context
import android.view.Gravity
import android.view.View
import android.view.WindowManager
import android.widget.TextView

import com.hy.frame.R
import com.hy.frame.common.BaseDialog

/**
 * 加载对话框
 * author HeYan
 * time 2015/12/11 18:11
 */
class LoadingDialog constructor(context: Context, private var loadMsg: String? = null) : BaseDialog(context) {

    private var txtLoadMsg: TextView? = null

    override fun initLayoutId(): Int = R.layout.dlg_loading
    
    override fun initWindow() {
        windowDeploy(WindowManager.LayoutParams.WRAP_CONTENT.toFloat(), WindowManager.LayoutParams.WRAP_CONTENT.toFloat(), Gravity.CENTER)
    }

    override fun initView() {
        txtLoadMsg = findViewById(R.id.loading_txtLoadMsg)
    }

    override fun initData() {
        if (loadMsg != null)
            txtLoadMsg!!.text = loadMsg
    }

    override fun onViewClick(v: View) {}

    fun updateMsg(msg: String) {
        if (txtLoadMsg != null) {
            txtLoadMsg!!.text = msg
        } else {
            loadMsg = msg
        }
    }

}
